### PR TITLE
8331285: Deprecate and obsolete OldSize

### PR DESCRIPTION
**Pull request by albertnetymk**
Simple deprecating a jvm flag.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change requires CSR request [JDK-8331286](https://bugs.openjdk.org/browse/JDK-8331286) to be approved

### Issues
 * [JDK-8331285](https://bugs.openjdk.org/browse/JDK-8331285): Deprecate and obsolete OldSize (**Enhancement** - P4)
 * [JDK-8331286](https://bugs.openjdk.org/browse/JDK-8331286): Deprecate and obsolete OldSize (**CSR**)

### Reviewers
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**)
 * [Erik Österlund](https://openjdk.org/census#eosterlund) (@fisk - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/18994/head:pull/18994` \
`$ git checkout pull/18994`

Update a local copy of the PR: \
`$ git checkout pull/18994` \
`$ git pull https://git.openjdk.org/jdk.git pull/18994/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 18994`

View PR using the GUI difftool: \
`$ git pr show -t 18994`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/18994.diff">https://git.openjdk.org/jdk/pull/18994.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/18994#issuecomment-2082345029)

### CODE REVIEW

File: src/hotspot/share/runtime/arguments.cpp [hunk 1]
```diff
@@ -504,6 +504,7 @@ static SpecialFlag const special_jvm_flags[] = {
   { "PreserveAllAnnotations",       JDK_Version::jdk(23), JDK_Version::jdk(24), JDK_Version::jdk(25) },
   { "UseNotificationThread",        JDK_Version::jdk(23), JDK_Version::jdk(24), JDK_Version::jdk(25) },
   { "UseEmptySlotsInSupers",        JDK_Version::jdk(23), JDK_Version::jdk(24), JDK_Version::jdk(25) },
+  { "OldSize",                      JDK_Version::jdk(23), JDK_Version::jdk(24), JDK_Version::jdk(25) },
```
Comment (dholmes-ora) [inline src/hotspot/share/runtime/arguments.cpp:507]: This should be at line 504 before PreserveAllAnnotations to keep ordering within release.

**Review by dholmes-ora — CHANGES_REQUESTED** [review]

**Review by dholmes-ora — APPROVED** [review]
Looks good. Thanks.

**Review by fisk — APPROVED** [review]
Looks good.

**Comment by albertnetymk** [comment]
Thanks for review.

/integrate

**Pull request closed without merging**